tests/containers/workbenches/jupyterlab/jupyterlab_test.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import requests
+
+import allure
+import pytest
+
+from tests.containers import docker_utils
+from tests.containers.workbenches.workbench_image_test import WorkbenchContainer, skip_if_not_workbench_image
+
+if TYPE_CHECKING:
+    import docker.models.images
+
+
+class TestJupyterLabImage:
+    """Tests for JupyterLab Workbench images in this repository."""
+
+    APP_ROOT_HOME = "/opt/app-root/src"
+
+    @allure.issue("RHOAIENG-11156")
+    @allure.description("Check that the HTML for the spinner is contained in the initial page.")
+    def test_spinner_html_loaded(self, image: str) -> None:
+        skip_if_not_jupyterlab_image(image)
+
+        container = WorkbenchContainer(image=image, user=4321, group_add=[0])
+        # if no env is specified, the image will run
+        # > 4321        3334    3319  0 10:36 pts/0    00:00:01 /mnt/rosetta /opt/app-root/bin/python3.11 /opt/app-root/bin/jupyter-lab
+        # > --ServerApp.root_dir=/opt/app-root/src --ServerApp.ip= --ServerApp.allow_origin=* --ServerApp.open_browser=False
+        # which does not let us open a notebook, we need to set
+        container.with_env("NOTEBOOK_ARGS", "\n".join([
+            "--ServerApp.port=8888",
+            "--ServerApp.token=''",
+            "--ServerApp.password=''",
+            "--ServerApp.base_url=/notebook/opendatahub/jovyan",
+            "--ServerApp.quit_button=False",
+            """--ServerApp.tornado_settings={"user":"jovyan","hub_host":"https://opendatahub.io","hub_prefix":"/notebookController/jovyan"}"""]))
+        try:
+            # we changed base_url, and wait_for_readiness=True would attempt connections to /
```

Comment:
Could you please clarify, why we need to override the default base URL here?

Reply:
because when jupyterlab is started with `--ServerApp.base_url=/notebook/opendatahub/jovyan`, accessing / will give you 404, and you need to go to /notebook/opendatahub/jovyan to get a 200 response

and starting jupyterlab without the jupyterlab options will not give you the page with spinner, but some welcome screen where it instructs you how to set tokens to connect

I guess I could've just disabled password and ignored all the other options, but I decided to use all the options dashboard normally starts the workbench with

Reply:
Hm, I see now; thanks for the explanation! It would be nice to put some brief explanation into the description too, if you don't mind; but I don't insist.

Reply:
That's reasonable, comment added.